packages/ruby/index.ts

```diff
@@ -47,30 +47,27 @@ async function bundleInstall(
 ) {
   debug(`running "bundle install --deployment"...`);
   const bundleAppConfig = await getWriteableDirectory();
+  const { exitCode, stderr, stdout } = await execa(
+    bundlePath,
+    ['install', '--deployment', '--gemfile', gemfilePath, '--path', bundleDir],
+    {
+      stdio: 'pipe',
+      reject: false,
+      env: {
+        ...process.env,
+        BUNDLE_SILENCE_ROOT_WARNING: '1',
+        BUNDLE_APP_CONFIG: bundleAppConfig,
+        BUNDLE_JOBS: '4',
+      },
+    }
+  );
 
-  try {
-    await execa(
-      bundlePath,
-      [
-        'install',
-        '--deployment',
-        '--gemfile',
-        gemfilePath,
-        '--path',
-        bundleDir,
-      ],
-      {
-        stdio: 'pipe',
-        env: {
-          BUNDLE_SILENCE_ROOT_WARNING: '1',
-          BUNDLE_APP_CONFIG: bundleAppConfig,
-          BUNDLE_JOBS: '4',
-        },
-      }
-    );
-  } catch (err) {
-    debug(`failed to run "bundle install --deployment"...`);
-    throw err;
+  if (exitCode === 0 || exitCode === 18) {
+    // Gemfile might contain "2.7.x" so install might exit with code 18 and message:
+    // "Your Ruby patchlevel is 0, but your Gemfile specified -1"
+    // See https://github.com/rubygems/bundler/blob/3f0638c6c8d340c2f2405ecb84eb3b39c433e36e/lib/bundler/errors.rb#L49
+  } else {
+    throw new Error(stdout + stderr);
```

Comment:
Do we care about the `exitCode` when there is a non-zero, non-18 exit code? Append a ` (code ${exitCode})` to the error message or set an `err.exitCode = exitCode`?

Reply:
Good point! I updated this message to look more like the previous message:

```
"bundle install" failed with exit code ${exitCode}
```